From ccdfd122cdaef77794fa23b54d242e98b5b732b8 Mon Sep 17 00:00:00 2001
From: Kuniyuki Iwashima <kuniyu@amazon.com>
Date: Mon, 24 Jul 2023 01:48:00 +0000
Subject: [PATCH] hsiphash

Signed-off-by: Kuniyuki Iwashima <kuniyu@amazon.com>
---
 include/net/ipv6.h | 11 +++++------
 1 file changed, 5 insertions(+), 6 deletions(-)

diff --git a/include/net/ipv6.h b/include/net/ipv6.h
index 7332296eca44..06463791b73a 100644
--- a/include/net/ipv6.h
+++ b/include/net/ipv6.h
@@ -749,15 +749,14 @@ static inline u32 ipv6_addr_hash(const struct in6_addr *a)
 #endif
 }
 
+static const hsiphash_key_t test_key_hsiphash = {{ 0x0706050403020100ULL, 0x0f0e0d0c0b0a0908ULL }};
+
 /* more secured version of ipv6_addr_hash() */
 static inline u32 __ipv6_addr_jhash(const struct in6_addr *a, const u32 initval)
 {
-	u32 v = (__force u32)a->s6_addr32[0] ^ (__force u32)a->s6_addr32[1];
-
-	return jhash_3words(v,
-			    (__force u32)a->s6_addr32[2],
-			    (__force u32)a->s6_addr32[3],
-			    initval);
+	return hsiphash_4u32(a->s6_addr32[0], a->s6_addr32[1],
+			     a->s6_addr32[2], a->s6_addr32[3],
+			     &test_key_hsiphash);
 }
 
 static inline bool ipv6_addr_loopback(const struct in6_addr *a)
-- 
2.40.1

